### ISSUE TITLE
PHPUnit 3.7.* dev releases create confusion in composer.json

### ISSUE THREAD

**Issue by tarjei** [sebastianbergmann/phpunit#852]
Hi, 

tl;dr
A lot of projects depend on phpunit 3.7.\* but the dependencies of phpunit wrt to the Symfony YAML component differ over versions.

I'm not sure if I see the complete picture or if I have the best solution, but I thought I'd file an issue anyway.

Since composer looks at the phpunit github repo for versions it will find branches that are not released (e.g. the 3.7.15 branch) and look at the dependencies in that branch.

This gave me a lot of trouble yesterday as 3.7.15 depends the 2.2.0 version of the YAML component and this dependency drags the rest of symfony with it up to 2.2. 

I actually ended up upgrading my project to 2.2 to get around my problems yesterday, but I think that if there are not specific features in YAML 2.2 that PHPUnit needs, then you should keep the dependency more lenient so that it is possible to use PHPUnit with older projects.
